framework/wazuh/event.py

```diff
@@ -0,0 +1,28 @@
+from wazuh.core.common import QUEUE_SOCKET
+from wazuh.core.results import WazuhResult
+from wazuh.core.wazuh_queue import WazuhAnalysisdQueue
+from wazuh.rbac.decorators import expose_resources
+
+MSG_HEADER = '1:API-Webhook:'
+
+
+@expose_resources(actions=["event:ingest"], resources=["*:*:*"], post_proc_func=None)
+def send_event_to_analysisd(events: list) -> WazuhResult:
+    """Send events to analysisd through the socket.
+
+    Parameters
+    ----------
+    events : list
+        List of events to send.
+
+    Returns
+    -------
+    WazuhResult
+        Confirmation message.
+    """
+
+    with WazuhAnalysisdQueue(QUEUE_SOCKET) as queue:
```

Comment:
@davidjiglesias I was working on the validation of the event size, but I have some doubts:

* The size should include the message header? (msg_header + msg) < 64Kb?
* If one event exceeds the limit which should be the behavior?  Return the `AffectedItemsWazuhResult` response maybe?

Reply:
Hi Nico, I can help with those questions:

> The size should include the message header? (msg_header + msg) < 64Kb?

Yes, 64KB includes the header size and the message. However, the size is not always 64KB, it could be changed by the user or in future releases. Maybe it should be handled in a similar way to wodles like gcloud or aws:
https://github.com/wazuh/wazuh/blob/e6819b2c2f37a9d5fe45873dcd9f6d1a24259445/wodles/aws/aws_s3.py#L356-L372

> If one event exceeds the limit which should be the behavior? Return the AffectedItemsWazuhResult response maybe?

Yes, returning AffectedItemsWazuhResult with the number of failed logs is a good option. Keep in mind that the endpoint must always answer with the same response format, so use `AffectedItemsWazuhResult` for both successful and failed responses.